src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -182,60 +184,46 @@ export class InstructorCoursesPageComponent implements OnInit {
           'Please make sure you have filled in both Course ID and Name before adding the course!');
       return;
     }
-    const paramMap: { [key: string]: string } = {
-      courseid: this.newCourseId,
-      coursename: this.newCourseName,
-      coursetimezone: this.timezone,
-      user: this.user,
-    };
-    this.newCourseId = '';
-    this.newCourseName = '';
-    this.timezone = moment.tz.guess();
-    this.httpRequestService.post('/instructor/courses', paramMap).subscribe((resp: MessageOutput) => {
-      this.loadInstructorCourses();
-      this.statusMessageService.showSuccessMessage(resp.message);
-    }, (resp: ErrorMessageOutput) => {
-      this.statusMessageService.showErrorMessage(resp.error.message);
-    });
-  }
 
-  /**
-   * Archives an active course.
-   */
-  onArchive(courseId: string): void {
-    if (!courseId) {
-      this.statusMessageService.showErrorMessage(`Course ${courseId} is not found!`);
-      return;
-    }
-    const paramMap: { [key: string]: string } = {
-      courseid: courseId,
-      archive: 'true',
-      user: this.user,
-    };
-    this.httpRequestService.put('/course', paramMap).subscribe((resp: MessageOutput) => {
+    this.courseService.createCourse({
+      courseName: this.newCourseName,
+      timeZone: this.timezone,
+      courseId: this.newCourseId,
+    }).subscribe((course: Course) => {
       this.loadInstructorCourses();
-      this.statusMessageService.showSuccessMessage(resp.message);
+      this.statusMessageService.showSuccessMessage('The course has been added. '
```

Comment:
I don't think the HTML will work. I believe the previous implementation also does not work. You can leave it here. If you know how to fix it. Go ahead.

Reply:
It doesn't? It seems to work just fine for me.
![image](https://user-images.githubusercontent.com/4904988/53417149-e86eb280-3a0f-11e9-95bb-65873320b249.png)
The link brings you correctly to the course enroll page and the edit course details page. 

Or is this not intended behavior?

Reply:
The HTML code is not being escaped 😕

Reply:
*was. The status message component has been extended to allow such case.